enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,901 @@
+        }
+    ]
+}
+$ opm index add --bundles quay.io/operatorhubio/etcd:v0.9.0 --mode replaces --config community-operators/etcd
+$ $ cat community-operators/etcd.json
+{
+    "schema": "olm.package",
+    "name": "etcd",
+    "defaultChannel": "singlenamespace-alpha",
+    "icon": {
+        "base64data":"iVBORw0KGgoAAAANSUhEUgAAA.....",
+        "mediatype":"image/png"
+    },
+    "channels": ["alpha", "singlenamespace-alpha"],
+    "description": "A message about etcd operator, a description of channels"
+},
+{
+    "schema": "olm.bundle", 
+    "name": "etcdoperator-community.v0.6.1",
+    "package": "etcd",
+    "image": "quay.io/operatorhubio/etcd:v0.6.1",
+    "version": "0.6.1",
+    "properties":[
+        {
+            "name": "olm.package",
+            "values": {
+                "packageName": "etcd",
+                "version": "0.6.1"
+            }
+        },
+        {
+            "name":"olm.gvk",
+            "values":{
+                "group": "etcd.database.coreos.com", 
+                "kind": "EtcdCluster", 
+                "version": "v1beta2"
+            }
+        },
+        {
+            "name":"olm.channel",
+            "value": "alpha"
+        }
+    ],
+    "relatedImages": [
+        {
+        "name": "etcdv0.6.1",
+        "image": "quay.io/coreos/etcd-operator@sha256:bd944a211eaf8f31da5e6d69e8541e7cada8f16a9f7a5a570b22478997819943"
+        }
+    ]
+},
+{
+    "schema": "olm.bundle",
+    "name": "etcdoperator.v0.9.0",
+    "package": "etcd",
+    "image": "quay.io/operatorhubio/etcd:v0.9.0",
+    "version": "0.9.0",
+    "properties":[
+        {
+            "name": "olm.package",
+            "values": {
+                "packageName": "etcd",
+                "version": "0.9.0"
+            }
+        },
+        {
+            "name":"olm.gvk",
+            "values":{
+                "group": "etcd.database.coreos.com", 
+                "kind": "EtcdBackup", 
+                "version": "v1beta2"
+            }
+        },
+        {
+            "name":"olm.channel",
+            "value": "singlenamespace-alpha"
+        }
+    ],
+    "relatedImages" : [
+        {
+            "name": "etcdv0.9.0",
+            "image": "quay.io/coreos/etcd-operator@sha256:db563baa8194fcfe39d1df744ed70024b0f1f9e9b55b5923c2f3a413c44dc6b8"
+        }
+    ]
+}
+```
+
+### Using the package json representation
+
+Once every package inside an index is represented using a json/yaml config file, the config files can be leveraged to inspect/update/validate the packages inside the index without needing to rebuild the index from scratch.  
+#### Inspecting the packages inside an index
+
+A new sub-command `inspect` will be introduced under `opm index`.
+
+When `opm index inspect` is summoned, the index image will be downloaded, and the json representations of the packages inside the index will be unfurled in a folder with the same name as the index. 
```

Comment:
Minor nit: I think this behavior, where we unpackage the configuration files within an index and create a directory that matches the index name needs to be introduced earlier in the enhancement.

Reply:
It is introduced in the first sentence of Story 1.